acstools/polarization_tools.py

```diff
@@ -90,116 +333,49 @@ def __init__(self, pol0, pol60, pol120, filter_name, detector, pav3):
-        r120 = pol_measurements[2] * correction_terms[2]
-
-        i = (2 / 3) * (r0 + r60 + r120)
-        q = (2 / 3) * ((2 * r0) - r60 - r120)
-        u = (2 / np.sqrt(3)) * (r60 - r120)
-
-        return i, q, u
+        self.stokes_i, self.stokes_q, self.stokes_u = calc_stokes(self.pol0, self.pol60, self.pol120,
+                                                                  c0=self.c0, c60=self.c60, c120=self.c120)
 
     def calc_polarization(self):
         """
         Calculate the polarization parameters (fractional polarization and position angle) for
         using attributes set at initialization.
         """
 
-        self.polarization = self._calc_fraction(self.stokes_i, self.stokes_q, self.stokes_u,
-                                                transmission_correction=self.transmission_correction)
-
-        self.angle = self._calc_theta(self.stokes_q, self.stokes_u, self.pav3)
-
-    def _calc_fraction(self, i, q, u, transmission_correction=1):
-        """
-        Method for determining the fractional polarization.
-
-        Parameters
-        ----------
-        i : float
-            Stokes I parameter.
-        q : float
-            Stokes Q parameter.
-        u : float
-            Stokes U parameter.
-        transmission_correction : float (Default = 1)
-            Correction factor to account for the leak of photons with
-            non-parallel electric field position angles. See Section 5.3.4
-            of the ACS Data Handbook.
-
-        Returns
-        -------
-        pol_fraction : float
-            Polarization fraction.
-        """
-
-        pol_fraction = np.sqrt(q**2 + u**2) / i
-        pol_fraction *= transmission_correction
-
-        return pol_fraction
-
-    def _calc_theta(self, q, u, pav3):
-        """
-        Calculate the position angle of the electric field vector given Stokes Q and U.
-        For ACS, it is also necessary to supply the position angle of the V3 axis of the
-        telescope (PA_V3 in the primary header) and the name of the ACS detector used.
-
-        Parameters
-        ----------
-        q : float
-            Stokes Q parameter.
-        u : float
-            Stokes U parameter.
-        pav3 : float
-            Position angle of the V3 axis in units of degrees. Found in ACS primary headers
-            with keyword PA_V3.
-
-        Returns
-        -------
-        theta : float
-            Position angle of the electric field vector in degrees.
-        """
-
-        chi = -38.2 if self.detector.lower() == 'wfc' else -69.4
-        theta = 0.5 * np.degrees(np.arctan2(u, q)) + pav3 + chi
-        theta -= 360 if theta > 360 else 0
+        self.polarization = calc_fraction(self.stokes_i, self.stokes_q, self.stokes_u,
+                                          transmission_correction=self.transmission_correction)
 
-        return theta
+        self.angle = calc_theta(self.stokes_q, self.stokes_u, self.detector, self.pav3) * units.degree
```

Comment:
If `calc_theta` returns Quantity, manually applying `units.degree` here is no longer necessary.